packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -240,15 +239,14 @@ void _writeHostApi(Indent indent, Api api, Root root) {
                     });
                   } else if (method.returnType.isVoid) {
                     indent.writeln(call);
-                    indent.writeln('wrapped["${Keys.result}"] = null');
+                    indent.writeln('wrapped.add(null)');
                   } else {
-                    indent.writeln('wrapped["${Keys.result}"] = $call');
+                    indent.writeln('wrapped.add($call)');
                   }
                 }, addTrailingNewline: false);
                 indent.add(' catch (exception: Error) ');
                 indent.scoped('{', '}', () {
-                  indent.writeln(
-                      'wrapped["${Keys.error}"] = wrapError(exception)');
+                  indent.writeln('wrapped.add(wrapError(exception))');
```

Comment:
I think you'll need to restructure the code here a little due to the mutable/non-mutable list difference for the non-error vs error case, and make the exception handler be:

```
reply.reply(wrapError(exception))
return
```

instead of using `wrapped` and falling through.

Reply:
had to do a little bit of a strange work around for this since the reply is sent at a different time when async. Might be worth taking a second look at to see if you have a cleaner solution.